omnisend/includes/SDK/V1/class-batch.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Omnisend Client
+ *
+ * @package OmnisendClient
+ */
+
+namespace Omnisend\SDK\V1;
+
+use Omnisend\Internal\Utils;
+use WP_Error;
+
+defined( 'ABSPATH' ) || die( 'no direct access' );
+
+/**
+ * Omnisend Batch class. It should be used with Omnisend Client.
+ *
+ */
+class Batch {
+	public const PRODUCTS_ENDPOINT   = 'products';
+	public const CATEGORIES_ENDPOINT = 'categories';
+	public const EVENTS_ENDPOINT     = 'events';
+	public const CONTACTS_ENDPOINT   = 'contacts';
+	public const POST_METHOD         = 'POST';
+	public const PUT_METHOD          = 'PUT';
+
+	private const REQUIRED_PROPERTIES = array(
+		'endpoint',
+		'items',
+		'method',
+	);
+	private const STRING_PROPERTIES   = array(
+		'endpoint',
+		'method',
+		'origin',
+	);
+	private const ARRAY_PROPERTIES    = array(
+		'items',
+	);
+	private const AVAILABLE_ENDPOINTS = array(
+		self::PRODUCTS_ENDPOINT,
+		self::CATEGORIES_ENDPOINT,
+		self::EVENTS_ENDPOINT,
+		self::CONTACTS_ENDPOINT,
+	);
+	private const AVAILABLE_METHODS   = array(
+		self::POST_METHOD,
+		self::PUT_METHOD,
+	);
+
+	/**
+	 * @var string $endpoint
+	 */
+	private $endpoint = null;
+
+	/**
+	 * @var array $items
+	 */
+	private $items = null;
+
+	/**
+	 * @var string $method
+	 */
+	private $method = null;
+
+	/**
+	 * @var string $origin
+	 */
+	private $origin = null;
+
+	/**
+	 * Validate batch properties.
+	 *
+	 * It ensures that required properties are set and that they are valid.
+	 *
+	 * @return WP_Error
+	 */
+	public function validate(): WP_Error {
+		$error = new WP_Error();
+		$error = $this->validate_properties( $error );
+
+		if ( $error->has_errors() ) {
+			return $error;
+		}
+
+		$error = $this->validate_values( $error );
+
+		return $error;
+	}
+
+	/**
+	 * Convert batch to array
+	 *
+	 * If batch is valid it will be transformed to array that can be sent to Omnisend.
+	 *
+	 * @return array
+	 */
+	public function to_array(): array {
+		if ( $this->validate()->has_errors() ) {
+			return array();
+		}
+
+		$arr = array(
+			'endpoint' => $this->endpoint,
+			'items'    => $this->items,
+			'method'   => $this->method,
+		);
+
+		if ( ! empty( $this->origin ) ) {
+			$arr['origin'] = $this->origin;
+		}
+
+		return $arr;
+	}
+
+	/**
+	 * Sets endpoint
+	 *
+	 * @param string $endpoint
+	 *
+	 * @return void
+	 */
+	public function set_endpoint( $endpoint ): void {
+		$this->endpoint = $endpoint;
+	}
+
+	/**
+	 * Sets batch items
+	 *
+	 * @param array $items
+	 *
+	 * @return void
+	 */
+	public function set_items( $items ): void {
```

Comment:
How are you planing to validate items? Because they can be product, category etc and as I understand batch methods will be used to create all of these items. 
How would you create a batch of products and send them via this method? Would to_array generate correct request?